Sources/EdgeNetworkHandlers/ResponseCallbackHandler.swift

```diff
@@ -39,32 +51,93 @@ class ResponseCallbackHandler {
+    ///
+    /// - Parameters:
+    ///   - requestEventId: unique event identifier for which the response callback is registered; should not be empty
+    ///   - completionHandler: the completion handler that needs to be registered, should not be nil
+    func registerCompletionHandler(requestEventId: String, completionHandler: (([EdgeEventHandle], [EdgeEventError]) -> Void)?) {
+        guard let unwrappedCompletion = completionHandler else { return }
+        guard !requestEventId.isEmpty else {
+            Log.warning(label: TAG, "Failed to register completion handler because of empty request event id.")
+            return
+        }
+
+        completionHandlers[requestEventId] = unwrappedCompletion
+    }
+
+    /// Calls onComplete and unregisters a `EdgeResponseHandler` for the specified `requestEventId`. After this operation,
+    /// the associated response handler or completion handler is removed and no longer called.
     /// - Parameter requestEventId: unique event identifier for experience events; should not be empty
-    func unregisterResponseHandler(requestEventId: String) {
+    func unregisterCallbacks(requestEventId: String) {
         guard !requestEventId.isEmpty else { return }
-
-        if responseHandlers[requestEventId] != nil {
+        if let responseHandler = responseHandlers[requestEventId] {
+            responseHandler.onComplete()
             responseHandlers[requestEventId] = nil
-            Log.trace(label: TAG, "Removing callback for Edge response with request unique id \(requestEventId).")
         }
+
+        if let completionHandler = completionHandlers[requestEventId] {
+            completionHandler(edgeEventHandles[requestEventId] ?? [], edgeEventErrors[requestEventId] ?? [])
+            completionHandlers[requestEventId] = nil
+        }
+
+        edgeEventHandles[requestEventId] = nil
+        edgeEventErrors[requestEventId] = nil
+
+        Log.trace(label: TAG, "Removing completion handlers for Edge response with request unique id \(requestEventId).")
+    }
+
+    /// Updates the list of `EdgeEventHandle`(s) for current `requestEventId` and calls onResponseUpdate if a `ResponseHandler` is registered.
+    /// - Parameters:
+    ///   - eventHandle: newly received event handle
+    ///   - requestEventId: the request event identifier associated with this event handle
+    func eventHandleReceived(_ eventHandle: EdgeEventHandle, requestEventId: String?) {
+        guard let unwrappedRequestEventId = requestEventId, !unwrappedRequestEventId.isEmpty else { return }
+        if edgeEventHandles[unwrappedRequestEventId] != nil {
+            edgeEventHandles[unwrappedRequestEventId]?.append(eventHandle)
+        } else {
+            edgeEventHandles[unwrappedRequestEventId] = [eventHandle]
+        }
+        invokeResponseHandler(eventHandle: eventHandle, eventError: nil, requestEventId: unwrappedRequestEventId)
```

Comment:
If this requestEventId is keyed to a completionHandler, and not a responseHandler, then calling `invokeResponseHandler` will log that the response handler was not found, even though event handler was correctly processed. Instead, can another conditional statement be added to check if the requestEventId is in the responseHandlers dictionary, and if so just call `invokeResponseHandler` otherwise add the eventHandle to edgeEventHandles dictionary?